src/core/server/elasticsearch/client/client_config.ts

```diff
@@ -96,6 +97,10 @@ export function parseClientOptions(
     );
   }
 
+  if (config.caFingerprint != null) {
+    clientOptions.caFingerprint = config.caFingerprint;
+  }
+
```

Comment:
NIT: as this doesn't need any transformation, you can put it in the initial config construction

```
  const clientOptions: ClientOptions = {
    sniffOnStart: config.sniffOnStart,
    sniffOnConnectionFault: config.sniffOnConnectionFault,
    caFingerprint: config.caFingerprint
    headers: {
      ...DEFAULT_HEADERS,
      ...config.customHeaders,
    },
  };
```

Reply:
Yep, makes sense, will make this change :+1:

Reply:
Done in https://github.com/elastic/kibana/pull/108514/commits/1cbf53d4a2ef7307cf433f203c5e1194007272bb

Reply:
Apparently, ES client treats `caFingerprint: undefined` as defined option :see_no_evil: https://github.com/elastic/elasticsearch-js/blob/a0dcace7cd2672a945b2333a245e199699a27789/index.js#L120

I'll change it to `caFingerprint: config.caFingerprint ?? null`

Reply:
> I'll change it to `caFingerprint: config.caFingerprint ?? null`

Well, that doesn't work either, type signature doesn't support `null`: https://github.com/elastic/elasticsearch-js/blob/2d1505eb2b7ad05387ca8ba8a6874f580a126ecd/lib/Connection.d.ts#L43

Sooo, it seems I have to revert it to how it was before :smile: